pkgs/applications/misc/bazecor/default.nix

```diff
@@ -17,34 +13,40 @@ appimageTools.wrapAppImage rec {
     # Workaround for https://github.com/Dygmalab/Bazecor/issues/370
     postExtract = ''
       substituteInPlace \
-        $out/usr/lib/bazecor/resources/app/.webpack/main/index.js \
-        --replace \
+        $out/usr/lib/${pname}/resources/app/.webpack/main/index.js \
+        --replace-fail \
           'checkUdev=()=>{try{if(c.default.existsSync(f))return c.default.readFileSync(f,"utf-8").trim()===l.trim()}catch(e){console.error(e)}return!1}' \
           'checkUdev=()=>{return 1}'
     '';
   };
+in appimageTools.wrapAppImage {
+  inherit pname version src;
 
   # also make sure to update the udev rules in ./10-dygma.rules; most recently
   # taken from
   # https://github.com/Dygmalab/Bazecor/blob/v1.3.11/src/main/utils/udev.ts#L6
 
-  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
-    p.glib
-  ];
+  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [ p.glib ];
 
   # Also expose the udev rules here, so it can be used as:
   #   services.udev.packages = [ pkgs.bazecor ];
   # to allow non-root modifications to the keyboards.
 
   extraInstallCommands = ''
-    install -m 444 -D ${src}/Bazecor.desktop -t $out/share/applications
-    substituteInPlace $out/share/applications/Bazecor.desktop \
-      --replace 'Exec=Bazecor' 'Exec=bazecor'
+    ln -s $out/bin/${pname}-${version} $out/bin/${pname}
+
+    source "${makeWrapper}/nix-support/setup-hook"
+    wrapProgram $out/bin/${pname} \
```

Comment:
I would place this behind the `enableWayland` flag as initially suggested, not sure if this would affect users on Xorg

Reply:
> I would place this behind the enableWayland flag as initially suggested

I thought about it a while ago and it really didn't sit right with me in the end to add a new variable to the build that we would have to maintain. I quickly looked around in the `nixpkgs` repo for toher uses of the `enableWayland` input and it was never used the way I intended to use it so it felt very icky.

I updated to `zig v0.3.0` recently and bazecor with the wayland flags never crashes on startup. Since Bazecor is still on `1.3.11` and has been for a while, it looks like the crashes were a compositor issue and not related to bazecor anyway. I think the best way going forward is to enable wayland support by default and users who experience problems on their side can overlay the `wrapProgram` command instead. No need for `enableWayland` anymore.


> not sure if this would affect users on Xorg

This PR has no effect on Xorg users since they won't have the `NIXOS_OZONE_WL` var set up in their `configuration.nix`. If `NIXOS_OZONE_WL` is not defined, it wraps the `bazecor` command with nothing.